fact.go

```diff
@@ -6,79 +6,86 @@ import (
 	"strconv"
 )
 
-// ErrInvalidNumericFact is returned when a numeric fact does not conform to the XBRL spec and a numeric value cannot be extracted from it.
-// For example, if it's a fraction type fact and the denominator is 0, or the fact is malformed and missing attributes.
-var ErrInvalidNumericFact = errors.New("numeric fact contains invalid data")
+type FactType string
 
-// NilFact is a fact in an XBRL document that has an xsi:nil set to a truthy value.
-// These facts still must have a contextRef, but nothing else is guaranteed.
-//
-// Detailed info on Facts in general can be found here:
-// https://www.xbrl.org/Specification/XBRL-2.1/REC-2003-12-31/XBRL-2.1-REC-2003-12-31+corrected-errata-2013-02-20.html#_4.6
-type NilFact struct {
-	XMLName xml.Name
+const (
+	// FactTypeNil is a fact which has an `xsi:nil` attribute set to a truthy value.
+	// A nil fact is only guaranteed to have an XMLName and ContextRef.
+	FactTypeNil FactType = "nil"
```

Comment:
I think these usually go the other way? `NilFactType`? Now I'm not sure.

Reply:
Oh really? I like it this way because it makes auto-complete easier. I can just type `xbrl.FactType` and the IDE will show me all the options. Although I imagine IDEs are smart enough now to do that even if it's named the other way..?

Reply:
I think I remembered it wrong.